### PR TITLE
Clickable breadcrumb for mimic game

### PR DESCRIPTION
**Pull request by yudino**
### Motivation

Make the breadcrumb clickable for the mimic game like in phase 1.

### Changes

In the files /creer-un-jeu/mimique/ : 
- 1.tsx
- 2.tsx
- 3.tsx
- 4.tsx

### Test

Go to phase 2 and click on the button "creer-un-jeu". Then in the next page, click on the button "Faire découvrir 3 mimiques".
You can test theses changes with the breadcrumb in the the top of the page.

### CODE REVIEW

File: src/pages/creer-un-jeu/mimique/2.tsx [hunk 1]
```diff
@@ -13,6 +13,15 @@ const MimiqueStep2 = () => {
...
   const { activity, updateActivity } = React.useContext(ActivityContext);
 
+  const data = (activity?.data as MimicsData) || null;
+
+  const isValid = React.useMemo(() => {
+    if (activity !== null && activity.content.filter((c) => c.value.length > 0 && c.value !== '<p></p>\n').length === 0) {
+      return false;
+    }
+    return true;
+  }, [activity]);
```
Comment (DavidRobertAnsart) [inline src/pages/creer-un-jeu/mimique/2.tsx:23]: I'm not sure the function here does what it should. You want to check if the step 1 of the mimic activity is valid. So:
- I would rename `isValid` to `isStep1Valid`.
- I would create a function in `/src/activity-types/game.constants.ts`: `function isMimicValid(data: MimicData) { [...] }` with the logic you have here: https://github.com/parlemonde/1village/blob/4c8fd297e57c842243443e957859997169e7bf4c/src/components/selectors/MimicSelector.tsx#L34 and use it: `return isMimicValid(data.game1)`

File: src/pages/creer-un-jeu/mimique/3.tsx [hunk 2]
```diff
@@ -13,6 +13,15 @@ const MimiqueStep3 = () => {
...
   const { activity, updateActivity } = React.useContext(ActivityContext);
 
+  const data = (activity?.data as MimicsData) || null;
+
+  const isValid = React.useMemo(() => {
+    if (activity !== null && activity.content.filter((c) => c.value.length > 0 && c.value !== '<p></p>\n').length === 0) {
+      return false;
+    }
+    return true;
+  }, [activity]);
```
Comment (DavidRobertAnsart) [inline src/pages/creer-un-jeu/mimique/3.tsx:23]: Same here, the goal would be to know whether step 1 is valid and whether step 2 is valid. Considering you have a function to check if a mimic is valid or not:
```tsx
const errorSteps = React.useMemo(() => {
    const errors: number[] = [];
    if (!isMimicValid(data.game1)) {
         errors.push(0); // step of mimic 1
    }
    if (!isMimicValid(data.game2)) {
         errors.push(1); // step of mimic 2
    }
    return errors;
  }, [data]);
```

File: src/pages/creer-un-jeu/mimique/4.tsx [hunk 3]
```diff
@@ -20,6 +21,13 @@ const MimiqueStep4 = () => {
   const [isLoading, setIsLoading] = React.useState(false);
   const data = (activity?.data as MimicsData) || null;
 
+  const isValid = React.useMemo(() => {
+    if (activity !== null && activity.content.filter((c) => c.value.length > 0 && c.value !== '<p></p>\n').length === 0) {
+      return false;
+    }
+    return true;
+  }, [activity]);
```
Comment (DavidRobertAnsart) [inline src/pages/creer-un-jeu/mimique/4.tsx:29]: And again here but these time with the 3 steps:
```tsx
const errorSteps = React.useMemo(() => {
    const errors: number[] = [];
    if (!isMimicValid(data.game1)) {
         errors.push(0); // step of mimic 1
    }
    if (!isMimicValid(data.game2)) {
         errors.push(1); // step of mimic 2
    }
    if (!isMimicValid(data.game3)) {
         errors.push(2); // step of mimic 3
    }
    return errors;
  }, [data]);
const isValid = errorSteps.length === 0;
```

**Review by DavidRobertAnsart — COMMENTED** [review]
There is only the validation functions to update: their goal is to validate previous steps, so you have to customize them based on the activity you are implementing.

Otherwise, the rest of the changes looks good to me!

File: src/pages/creer-un-jeu/mimique/2.tsx (hunk 1 above)
Comment (yudino) [reply src/pages/creer-un-jeu/mimique/2.tsx]: I did the change for this file and created a function in game.constants.ts as suggested.

File: src/pages/creer-un-jeu/mimique/3.tsx (hunk 2 above)
Comment (yudino) [reply src/pages/creer-un-jeu/mimique/3.tsx]: I made the change here too.

File: src/pages/creer-un-jeu/mimique/4.tsx (hunk 3 above)
Comment (yudino) [reply src/pages/creer-un-jeu/mimique/4.tsx]: I made the change here too as suggested. I changed also the logical in the files to adapt the cards regarding the validation or not. It seems to work.

**Review by DavidRobertAnsart — APPROVED** [review]
👍

**Pull request merged**